src/System.Memory/src/System/Buffers/MemoryHandle.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+using System.Runtime.InteropServices;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// A handle for the array.
+    /// </summary>
+    public unsafe struct MemoryHandle : IDisposable
+    {
+        private IRetainable _owner;
+        private void* _pointer;
+        private GCHandle _handle;
+
+        /// <summary>
+        /// Creates a new memory handle for the array.
+        /// </summary>
+        /// <param name="owner">The object owner which is responsible for releasing the object on dispose.</param>
+        /// <param name="pinnedPointer">If the object is pinned, this is its address. If unspecified, it is set to null</param>
+        /// <param name="handle">Used to release the handle for the array on dispose. If unspecified, it is set to default(GCHandle)</param>
```

Comment:
These 3 are not worded great. But I think to fix that, we first need to write remarks that explain the architecture. Please file an issue for that and for not simplify the parmameter docs to:
owner: reference to manually managed object
pinnedPointer: pointer to the buffer, or null if the buffer is not pinned.
handle: handle used to pin array buffers.

Reply:
Created an issue: https://github.com/dotnet/corefx/issues/23863